core/src/io/anuke/mindustry/core/Control.java

```diff
@@ -151,19 +159,29 @@ public Control(){
                     state.stats.buildingsDeconstructed++;
                 }else{
                     state.stats.buildingsBuilt++;
+                    state.stats.resourcesSpent += Array.with(e.tile.block().buildRequirements).sum(itemStack -> itemStack.amount);
```

Comment:
`Array.with` and `Array.select` both allocate new arrays, which should be avoided unless you're running the code very rarely. In this case, the code is run every time a block is built, so it would be a better idea to just use a normal for loop.